test/Microsoft.AspNet.Mvc.Core.Test/ModelBinding/TypeMatchModelBinderTest.cs

```diff
@@ -0,0 +1,117 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.Framework.Internal;
+using Xunit;
+
+namespace Microsoft.AspNet.Mvc.ModelBinding.Test
+{
+    public class TypeMatchModelBinderTest
+    {
+        [Fact]
+        public async Task BindModel_InvalidValueProviderResult_ReturnsFalse()
+        {
+            // Arrange
+            var bindingContext = GetBindingContext();
+            bindingContext.ValueProvider = new SimpleHttpValueProvider
+            {
+                { "theModelName", "not an integer" }
+            };
+
+            var binder = new TypeMatchModelBinder();
+
+            // Act
+            var result = await binder.BindModelAsync(bindingContext);
+
+            // Assert
+            Assert.Null(result);
+            Assert.Empty(bindingContext.ModelState);
```

Comment:
shouldn't there be a conversion error here?

Would like to see much more in depth verification of the model state.